src/neural/network_opencl.cc

```diff
@@ -134,111 +134,102 @@ class OpenCLNetwork : public Network {
-     static constexpr int NUM_POLICY_INPUT_PLANES = 32;
-     static constexpr int NUM_OUTPUT_POLICY = 1858;
-     static constexpr int NUM_VALUE_CHANNELS = 128;
-     */
+    ////////////////////////////////////////////////////////////////////////////
+    // Input convolution
 
-    int NUM_VALUE_INPUT_PLANES = weights.value.bn_means.size();
-    int NUM_POLICY_INPUT_PLANES = weights.policy.bn_means.size();
-    int NUM_OUTPUT_POLICY = weights.ip_pol_b.size();
-    int NUM_VALUE_CHANNELS = weights.ip1_val_b.size();
-
-    static constexpr auto kWinogradAlpha = 4;
-
-    opencl_.initialize(channels, params_);
+    opencl_.initialize(weights.input.biases.size(), params_);
 
     auto tuners = opencl_.get_sgemm_tuners();
 
     auto mwg = tuners[0];
     auto kwg = tuners[2];
     auto vwm = tuners[3];
 
-    size_t m_ceil = ceilMultiple(ceilMultiple(channels, mwg), vwm);
-    size_t k_ceil = ceilMultiple(ceilMultiple(inputChannels, kwg), vwm);
+    size_t m_ceil = ceilMultiple(ceilMultiple(weights.input.biases.size(), mwg), vwm);
+    size_t k_ceil = ceilMultiple(ceilMultiple(kInputPlanes, kwg), vwm);
 
-    std::vector<float> input_conv_weights = Transforms::WinogradTransformF(
-        weights.input.weights, channels, inputChannels);
+    std::vector<float> input_conv_weights = BlasTransforms::WinogradTransformF(
+        weights.input.weights, weights.input.biases.size(), kInputPlanes);
 
-    auto Upad = Transforms::ZeropadU(input_conv_weights, channels,
-                                     inputChannels, m_ceil, k_ceil);
+    auto Upad = BlasTransforms::ZeropadU(input_conv_weights, weights.input.biases.size(),
+                                     kInputPlanes, m_ceil, k_ceil);
 
     std::vector<float> input_batchnorm_means =
         weights.input.bn_means;  // copy ctor
-    Transforms::OffsetBatchNormMeans(input_batchnorm_means,
+    BlasTransforms::OffsetBatchNormMeans(input_batchnorm_means,
                                      weights.input.biases);
 
     std::vector<float> input_batchnorm_stddivs = weights.input.bn_stddivs;
-    Transforms::InvertBatchNormStddev(input_batchnorm_stddivs);
+    BlasTransforms::InvertBatchNormStddev(input_batchnorm_stddivs);
 
     // Winograd filter transformation changes filter size to 4x4
-    opencl_net_.push_input_convolution(kWinogradAlpha, inputChannels, channels,
+    opencl_net_.push_input_convolution(kWinogradAlpha, kInputPlanes, weights.input.biases.size(),
                                        Upad, input_batchnorm_means,
                                        input_batchnorm_stddivs);
 
-    // residual blocks
-    for (auto i = size_t{0}; i < residual_blocks; i++) {
-      auto& residual = weights.residual[i];
+    ////////////////////////////////////////////////////////////////////////////
+    // Residual tower
+
+    for (auto& residual : weights.residual) {
       auto& conv1 = residual.conv1;
       auto& conv2 = residual.conv2;
 
       std::vector<float> conv_weights_1 =
```

Comment:
Watch the code you changed. Previous version versus newest Version.
And ask you the question: which one the more easy to read?

 `Transforms::WinogradTransformF(conv1.weights, channels, channels);`

or 

 `BlasTransforms::WinogradTransformF(conv1.weights, conv1.biases.size(), conv1.biases.size());`

Reply:
The latter is far easier to read, since it doesn't have pointless redirection and obfuscation.

If `channels` had some actual calculation going on, such that it differed from `biases.size()` in some way, then I'd say sure, lets make a separate variable. But if nothing is actually changing, then adding a separate variable only adds confusion.

Reply:
No, of course not. And you probably know yourself that this is not true. For someone you is not familiar with the code, and even for me, the first version is far much more readable.
Do you realise that you oppose every comment that is made to you? You are not doing any single compromise. What the point of reviewing?